modules/platforms/dotnet/Apache.Ignite.Core.Tests/Client/ClientConnectionTest.cs

```diff
@@ -1045,6 +1045,35 @@ public void TestSendServerExceptionStackTraceToClient()
             StringAssert.Contains("at org.apache.ignite.platform.PlatformComputeEchoTask$EchoJob.execute", ex.Message);
         }
 
+        /// <summary>
+        /// Tests that server rejects new connections when the connection limit reached.
+        /// </summary>
+        [Test]
+        public void TestRejectOnClientLimitReached()
+        {
+            const int MAX_CONNECTIONS = 4;
+            
+            var ignite = Ignition.Start(new IgniteConfiguration(TestUtils.GetTestConfiguration())
+            {
+                ClientConnectorConfiguration = new ClientConnectorConfiguration
+                {
+                    MaxConnections = MAX_CONNECTIONS
+                }
+            });
+
+            Assert.AreEqual(MAX_CONNECTIONS, ignite.GetConfiguration().ClientConnectorConfiguration.MaxConnections);
+
+            var clients = new IIgniteClient[MAX_CONNECTIONS];
+            for (var i = 0; i < MAX_CONNECTIONS; ++i)
+            {
+                clients[i] = StartClient();
+            }
+
+            var ex = Assert.Catch<Exception>(() => StartClient());
+
+            StringAssert.Contains("Connection limit reached: " + MAX_CONNECTIONS, ex.Message);
```

Comment:
We should close (dispose) the clients after the test ends:

```suggestion
            var clients = new List<IIgniteClient>();

            try
            {
                clients.AddRange(Enumerable.Range(1, 10).Select(_ => StartClient()));

                var ex = Assert.Catch<Exception>(() => StartClient());
                StringAssert.Contains("Connection limit reached: " + maxConnections, ex.Message);
            }
            finally
            {
                clients.ForEach(c => c.Dispose());
            }
```

Reply:
Fixed